### ISSUE TITLE
Panic while using conrod in Fedora 25 (using Wayland)

### ISSUE THREAD

**Issue by AaronM04** [tomaka/winit#97]
Hi, I started a conrod app while running Wayland, and the window appeared, but as soon as I moved the mouse over the window, it panicked:

```
$ cargo run --bin client
    Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/client`
wl_display@1: error 1: invalid method 9 (since 3 < 4), object wl_surface@25
thread 'main' panicked at 'Wayland connection unexpectedly lost: Error { repr: Os { code: 22, message: "Invalid argument" } }', ../src/libcore/result.rs:837
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', ../src/libcore/result.rs:837
stack backtrace:
   1:     0x55cdf5cc0fea - std::sys::imp::backtrace::tracing::imp::write::he3d1bfbdbf113480
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x55cdf5cc51af - std::panicking::default_hook::{{closure}}::h575f1b40d2e88f07
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:247
   3:     0x55cdf5cc34c6 - std::panicking::default_hook::h3d5dccce8125d0cf
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:263
   4:     0x55cdf5cc3b77 - std::panicking::rust_panic_with_hook::h00b81bb3dcbd51f2
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:451
   5:     0x55cdf5cc3a04 - std::panicking::begin_panic::ha6a0d553db9869ff
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:413
   6:     0x55cdf5cc3929 - std::panicking::begin_panic_fmt::h24d113aee3ee4081
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:397
   7:     0x55cdf5cc38b7 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:373
   8:     0x55cdf5cff73d - core::panicking::panic_fmt::he441b2ea2036b98a
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/panicking.rs:69
   9:     0x55cdf5be64fe - core::result::unwrap_failed::h327f68929ca5f690
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/macros.rs:29
  10:     0x55cdf5bdae80 - <core::result::Result<T, E>>::unwrap::he114e0685fd77f9f
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libcore/result.rs:737
  11:     0x55cdf5bfb538 - winit::api::wayland::context::WaylandContext::prune_dead_windows::h0043fbe011f3d42e
                        at /home/aaron/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.5.5/src/api/wayland/context.rs:274
  12:     0x55cdf5c00f3f - <winit::api::wayland::window::Window as core::ops::Drop>::drop::h609c06892476c0ad
                        at /home/aaron/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.5.5/src/api/wayland/window.rs:296
  13:     0x55cdf58f0ba0 - drop::h53d13dfd33f73a71
  14:     0x55cdf58effc7 - drop::h3731c4444e92ec7f
  15:     0x55cdf58f17f8 - drop::h7c7eb2fb5546c331
  16:     0x55cdf58f3b90 - drop::hebc6ff9f4701a924
  17:     0x55cdf58f2838 - drop::haed05670111a409f
  18:     0x55cdf58f2930 - drop::hb4156e731b1ea966
  19:     0x55cdf58f2f70 - drop::hc5482f52e0a2442b
  20:     0x55cdf5983205 - client::main::h1b9234f28a4a1eba
                        at /home/aaron/dev/rust/conway/ConWaysteTheEnemy/src/client.rs:65
  21:     0x55cdf5cccc8a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libpanic_unwind/lib.rs:97
  22:     0x55cdf5cc4595 - std::rt::lang_start::h74b3afbdd8daef1c
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:332
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/rt.rs:57
  23:     0x55cdf59a5113 - main
  24:     0x7f6a199a5400 - __libc_start_main
  25:     0x55cdf5887998 - _start
  26:                0x0 - <unknown>
thread panicked while panicking. aborting.
```

Is it a problem with this crate, or something else? Thanks! :)

**Comment by tomaka** [comment]
Cc @vberger

**Comment by vberger** [comment]
Okay, this is awfully similar to https://github.com/vberger/wayland-client-rs/issues/75

**Comment by vberger** [comment]
@AaronM04 I suggest we continue this on the issue on my repo wayland-client-rs.

Would you by chance have a small code reproducing the error ?

**Comment by AaronM04** [comment]
I actually just shipped my laptop in for battery repairs so I can't test it anymore. Try cloning the conrod repo at https://github.com/pistondevelopers/conrod.git and running the old_demo example: `cargo run --example old_demo`

If that doesn't panic for you, I'll look in my backups for the code I used.

On December 5, 2016 1:29:44 PM PST, Victor Berger <notifications@github.com> wrote:
>@AaronM04 I suggest we continue this on the issue on my repo
>wayland-client-rs.
>
>Would you by chance have a small code reproducing the error ?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/tomaka/winit/issues/97#issuecomment-264983068

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by vberger** [comment]
Doesn't panic for me.

But if it is indeed the same issue as the other bug, it's caused by some subtlety of the system libraries. Apparently some combination of mesa and gnome on fedora 25 makes something go wrong.

Are you using gnome, by chance?

**Comment by AaronM04** [comment]
Yes, Gnome 3.22

On December 5, 2016 2:01:05 PM PST, Victor Berger <notifications@github.com> wrote:
>Doesn't panic for me.
>
>But if it is indeed the same issue as the other bug, it's caused by
>some subtlety of the system libraries. Apparently some combination of
>mesa and gnome on fedora 25 makes something go wrong.
>
>Are you using gnome, by chance?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/tomaka/winit/issues/97#issuecomment-264991748

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by nical** [comment]
This was just fixed in wayland-window 0.4.3,  see: https://github.com/vberger/wayland-client-rs/issues/75

**Comment by vberger** [comment]
This should have been closed long ago
